engine/flink/components/base-unbounded/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/SampleGeneratorSourceFactory.scala

```diff
@@ -88,6 +89,14 @@ class SampleGeneratorSourceFactory(timestampAssigner: TimestampWatermarkHandler[
     }
   }
 
+  class ContextMultiplierPerCountFunction(count: Int) extends FlatMapFunction[Context, Context] with Serializable {
```

Comment:
I think that we should create a new context for each invocation as well. Thanks to that it be possible to correlate each generated value with things that happened in the scenario

Reply:
The context we create at this level is actually only used for evaluation of the `value` parameter. Its simply just a temporary context just for that and its not used later. Later, a proper context is created in `FlinkContextInitializingFunction`.

I added a comment and I also made this source extend StandardFlinkSource instead of FlinkSource and removed some duplicate logic through that.

Reply:
Oh I see, we could extend `FlinkSource` instead of `StandardFlinkSource` and control when the context is generated. This temporary context id will be visible in errors topic and in test mechanism if we add a support for that. I'm not forcing to do it because current solution is also fine for now but if you want you can change it that way.

Reply:
Unfortunately, couldn't manage to use this approach to make it cleaner. Ultimately, the contexts are created per each event, so it should work regardless with current approach.